src/agent_sdk/inc/du_agent_sdk/agent_module_interface.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file agent_module_interface.h
+ * @brief Defines types and methods for DU Agent module library.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef ADUC_AGENT_MODULE_INTERFACE_H
+#define ADUC_AGENT_MODULE_INTERFACE_H
+
+#include "du_agent_sdk/agent_common.h"
+#include "aduc/result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+typedef void* ADUC_AGENT_MODULE_HANDLE;
+
+typedef ADUC_AGENT_MODULE_HANDLE (*ADUC_AGENT_MODULE_CREATE)(void);
+typedef void (*ADUC_AGENT_MODULE_DESTROY)(ADUC_AGENT_MODULE_HANDLE);
+
+typedef struct ADUC_AGENT_CONTRACT_INFO_TAG
+{
+    const char* provider;
+    const char* name;
+    const int version;
+    const char* contractInfo;
+} ADUC_AGENT_CONTRACT_INFO;
+
+/**
+ * @brief The module data types
+ *
+ */
+typedef enum ADUC_MODULE_DATA_TYPE_TAG
+{
+    ADUC_ModuleDataType_None = 0,   /**< No module data*/
+    ADUC_ModuleDataType_String = 1, /**< String data */
+    ADUC_ModuleDataType_Json = 2,   /**< JSON string data */
+    ADUC_ModuleDataType_Binary = 3, /**< Binary data */
```

Comment:
why are all these module data types needed? Could it not just standardize on one?